frontend/static/scripts/main.js

```diff
@@ -86,19 +86,85 @@ function gen_counterexamples_html(counterexamples){
 
 $(function () {
 
-    if (!Cookies.get('username')) {
-        $('#myModal').modal('show');
+    if (!Cookies.get('token')) {
+        $('#myModal').modal({backdrop: 'static', keyboard: false});
     }
 
-    $('#myModal').on('hidden.bs.modal', function (e) {
-        if ($('#username-input').val() == "" || $('#institution-input').val() == "") {
-            $('#myModal').modal('show');
+    $('.register-btn').click(function() {
+        var name = $('#name-input').val();
+        var email = $('#email-input').val();
+        var pass = $('#pass-input').val();
+        var inst = $('#institution-input').val();
+
+        if (!name || !email || !pass || !inst) {
+            return false;
```

Comment:
It would be nice if there is some hint to the user to show why this registration failed.